src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemMulticastIPAddressInformation.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net.NetworkInformation
+{
+    // Specifies the Multicast addresses for an interface.
+    internal class SystemMulticastIPAddressInformation : MulticastIPAddressInformation
+    {
+        private SystemIPAddressInformation _innerInfo;
+
+        private SystemMulticastIPAddressInformation()
+        {
+        }
```

Comment:
I believe it is to force you to call the factory method below (ToMulticastIpAddressInformationCollection(IPAddressInformationCollection addresses))